### PR TITLE
Import cannot be used after `::` so can be removed

### PR DESCRIPTION
**Pull request by SpacePossum**
closes https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4582

### CODE REVIEW

File: src/Fixer/Import/NoUnusedImportsFixer.php [hunk 1]
```diff
@@ -145,18 +145,20 @@ private function isImportUsed(Tokens $tokens, NamespaceAnalysis $namespace, arra
...
                 continue;
             }
 
-            if ($token->isComment() && Preg::match(
-                '/(?<![[:alnum:]])(?<!\\\\)'.$shortName.'(?![[:alnum:]])/i',
-                $token->getContent()
-            )) {
+            if ($token->isComment()
+                && Preg::match(
+                    '/(?<![[:alnum:]])(?<!\\\\)'.$shortName.'(?![[:alnum:]])/i',
```
Comment (roukmoute) [inline src/Fixer/Import/NoUnusedImportsFixer.php]: Remove `/i`.

**Comment by roukmoute** [comment]
I think you could add a new test:

```php
'in_same_namespace_3' => [
            <<<'EOF'
<?php

namespace Foo\Baz;

// My bar task.

EOF
            ,
            <<<'EOF'
<?php

namespace Foo\Baz;

use Foo\Bar;

// My bar task.

EOF
            ,
        ]
```

File: src/Fixer/Import/NoUnusedImportsFixer.php (hunk 1 above)
Comment (keradus) [reply src/Fixer/Import/NoUnusedImportsFixer.php]: the regex remain the same as it was. php imports are case insensitive.

Comment (GrahamCampbell) [reply src/Fixer/Import/NoUnusedImportsFixer.php]: > php imports are case insensitive.

Not necessarily, actually. When it comes to autoloading, case matters if one has a case sensitive filesystem. So, while class fully qualified names are case insensitive, if the autoloader is later fired, case matters.

**Comment by keradus** [comment]
Thank you @SpacePossum.

**Pull request merged**